packages/components/src/components/rating/rating.jsx

```diff
@@ -0,0 +1,134 @@
+    };
+
+    const handleMouseMove = e => {
+        if (readonly || disable_hover) {
+            return;
+        }
+        setIsHoverEnabled(true);
+        setHoveredRateIndex(calculateRating(e));
+    };
+
+    const handleMouseLeave = () => {
+        if (readonly || disable_hover) {
+            return;
+        }
+        setHoveredRateIndex(-1); // Reset to default state
+        setIsHoverEnabled(false);
+    };
+
+    const validateInitialValue = () => {
+        if (value < 0 || value > max_rating) {
+            setEnabledRateIndex(0);
+        } else {
+            const set_value = allow_half_rating ? parseFloat(value) : parseInt(value);
+            setEnabledRateIndex(set_value);
+        }
+    };
+
+    React.useEffect(() => {
+        validateInitialValue();
+    }, [value, max_rating]);
+
+    if (!!display_default_content && value <= 0) {
+        return display_default_content;
+    }
+
+    return (
+        <div className='user-rating'>
+            <div
+                className={classNames('user-rating__wrapper', background_class)}
+                style={{ direction }}
+                ref={ratingContainerRef}
+                onClick={handleClick}
+                onMouseMove={handleMouseMove}
+                onMouseLeave={handleMouseLeave}
+            >
+                {[...new Array(parseInt(max_rating))].map((_, index) => {
+                    const active_rate = is_hover_enabled ? hovered_rate_index : enabled_rate_index;
+                    const show_empty_icon = active_rate === -1 || active_rate < index + 1;
+                    const is_rating_with_precision = active_rate % 1 !== 0;
+                    const is_rating_equal_to_index = Math.round(active_rate) === index + 1;
+                    const show_rating_with_precision = is_rating_with_precision && is_rating_equal_to_index;
+                    return (
+                        <div className={classNames('user-rating__element', element_class)} key={index}>
+                            <section
+                                className={classNames('user-rating__element--filled')}
+                                style={{
+                                    width: show_rating_with_precision ? `${(active_rate % 1) * 100}%` : '0%',
+                                }}
+                            >
+                                {icon_selected}
+                            </section>
+                            <section>{show_empty_icon ? icon_unselected : icon_selected}</section>
+                        </div>
+                    );
+                })}
+            </div>
+            {show_rating && <Text>{parseFloat(enabled_rate_index).toFixed(1)}</Text>}
+        </div>
+    );
+};
+
+Rating.propTypes = {
+    allow_half_rating: PropTypes.bool,
+    background_class: PropTypes.string,
+    direction: PropTypes.oneOf(['ltr', 'rtl']),
+    disable_hover: PropTypes.bool,
+    display_default_content: PropTypes.element,
```

Comment:
can we rename it to `default_content`? 
`display_default_content` sounds like a boolean

Reply:
ok. I will rename it